pointcloud/basic_octree.go

```diff
@@ -182,3 +214,63 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// CollidesWith checks if the given octree collides with the given geometry and returns true if it does.
+func (octree *BasicOctree) CollidesWith(geom spatialmath.Geometry) (bool, error) {
+	return octree.CollidesWithGeometry(geom, confidenceThreshold, buffer)
+}
+
+// DistanceFrom returns the distance from the given octree to the given geometry.
+func (octree *BasicOctree) DistanceFrom(geom spatialmath.Geometry) (float64, error) {
+	// TODO (RSDK-3743): currently implemented as the bare minimum but needs to be changed to correct implementation
+	collides, err := octree.CollidesWith(geom)
+	if err != nil {
+		return math.Inf(1), err
+	}
+	if collides {
+		return -1, nil
+	}
+	return 1, nil
+}
+
+// EncompassedBy returns true if the given octree is within the given geometry.
+func (octree *BasicOctree) EncompassedBy(geom spatialmath.Geometry) (bool, error) {
+	// TODO (RSDK-3743)
+	return false, errors.New("not implemented")
```

Comment:
[q/nit]: here we use `errors.New()` but in `services/motion/builtin` we do `fmt.Errorf`, is there a reason to use `errors.New()` here?

Reply:
I'm not sure if there's a specific way to do it since I've seen both. I've been using `errors.New()` for errors that are just strings with no formatting required and `fmt.Errorf` for errors where I want to format with some variables. It seems like the errors package we import doesn't have a formatting option.

Reply:
> using errors.New() for errors that are just strings with no formatting required and fmt.Errorf for errors where I want to format with some variables

This is the correct usage; keep doing this :)